extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -138,25 +140,21 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
   }
 
   /** @override */
-  handleResize(width, height) {
-    const pWidth = this.element.getAttribute('width');
-    const pHeight = this.element.getAttribute('height');
-    // We want to resize only if neither returned dimension is larger than its
-    // primary counterpart, and if at least one of the returned dimensions
-    // differ from its primary counterpart.
-    if ((width != pWidth || height != pHeight)
-        && (width <= pWidth && height <= pHeight)) {
-      this.attemptChangeSize(height, width).catch(() => {});
-    }
-  }
-
-  /** @override */
   extractCreativeAndSignature(responseText, responseHeaders) {
     setGoogleLifecycleVarsFromHeaders(responseHeaders, this.lifecycleReporter_);
     this.ampAnalyticsConfig =
       extractAmpAnalyticsConfig(responseHeaders, this.extensions_);
     this.responseHeaders_ = responseHeaders;
-    return extractGoogleAdCreativeAndSignature(responseText, responseHeaders);
+    const adResponsePromise =
+        extractGoogleAdCreativeAndSignature(responseText, responseHeaders);
+    return adResponsePromise.then(adResponse => {
+      if (adResponse.size) {
+        this.handleResize_(adResponse.size.width, adResponse.size.height);
```

Comment:
handleResize could fail if the slot is within the viewport so I assume we want to also set on adResponse.size to ensure it would be centered in this case?